### PR TITLE
Add problem matcher for PHPUnit

### PR DESCRIPTION
**Pull request by decnorton**
---
name:  🎉 Add problem matcher for PHPUnit
about: Add support for parsing PHPUnit output using problem matches
labels: enhancement
---

### Description

Adds a problem matcher, which parses output for warnings/errors. Documentation can be found here: https://github.com/actions/toolkit/blob/master/docs/problem-matchers.md

The GitHub provided `setup-*` actions provide, [example](https://github.com/actions/setup-node/tree/7a3ce8362699742a34a1590e93c9b75e4d0d6dff/.github)

Matchers are registered using [toolkit commands](https://github.com/actions/toolkit/blob/master/docs/commands.md#problem-matchers). I've followed the implementation used by [`setup-node`](https://github.com/actions/setup-node/blob/7a3ce8362699742a34a1590e93c9b75e4d0d6dff/src/setup-node.ts#L27-L35)

---

> In case this PR introduced TypeScript/JavaScript code changes:

- [x] I have written test cases for the changes in this pull request
- [x] I have run `npm run format` before the commit.
- [x] `npm test` returns with no unit test errors.

### CODE REVIEW

File: .github/matchers/phpunit.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "problemMatcher": [
+    {
+      "owner": "phpunit",
+      "pattern": [
+        {
+          "regexp": "^\\d+\\)\\s.*$"
```
Comment (shivammathur) [inline .github/matchers/phpunit.json]: Add a test for this regular expression `^\\d+\\)\\s.*$` against PHPUnit output in `matchers.test.ts`

File: .github/matchers/phpunit.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        },
+        {
+          "regexp": "^(.*)$",
```
Comment (shivammathur) [inline .github/matchers/phpunit.json]: Add a test for this regular expression `^(.*)$` against PHPUnit output in `matchers.test.ts`

File: .github/matchers/phpunit.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+          "message": 1
+        },
+        {
+          "regexp": "^\\s*$"
```
Comment (shivammathur) [inline .github/matchers/phpunit.json]: Add a test for this regular expression `^\\s*$` against PHPUnit output in `matchers.test.ts`

File: .github/matchers/phpunit.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        },
+        {
+          "regexp": "^(.*):(\\d+)$",
```
Comment (shivammathur) [inline .github/matchers/phpunit.json]: Add a test for this regular expression `^(.*):(\\d+)$` against PHPUnit output in `matchers.test.ts`

**Comment by shivammathur** [comment]
@decnorton will you work on this PR this week, or I can merge this and and work on this further if you can give me some idea about each regex.

**Comment by decnorton** [comment]
@shivammathur Apologies, I didn't get a notification for your reply!

I will try and find some time this week to write the tests, but here's a brief explanation below:

Given the following output from PHPUnit:
```
1) Tests\Test::it_tests
Failed asserting that false is true.

/Users/decnorton/problem-matcher/tests/Test.php:42
```

The problem matcher will look for the following patterns as defined in the pattern array. Each line must be matched in order to be extracted.

`{ "regexp": "^\\d+\\)\\s.*$" }` matches `1) Tests\Test::it_tests`
The name of the failing test, we do not extract any information from this, as GitHub Actions will add an annotation at the correct location with the message (extracted below).

`{ "regexp": "^(.*)$", "message": 1 }` matches `Failed asserting that false is true.`
Extract the error message which can contain any anything. The message is in capture group 1.

`{ "regexp": "^\\s*$" }`
PHPUnit adds a blank line between the message and the file/line number. Match this and don't extract anything.

`{ "regexp": "^(.*):(\\d+)$", "file": 1, "line": 2 }` matches `/Users/decnorton/problem-matcher/tests/Test.php:42`
Extract the file and line number from the fourth line. File is in capture group 1, line is in capture group 2.

**Pull request merged**